csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -72,6 +74,22 @@ class AllocationOrderInferencer : public IterVisitor {
     return false;
   }
 
+  // returns the candidate that dominates the allocation order.
+  //
+  // It scans through each candidate to find the first one that:
+  //   1. is a TensorView
+  //   2. has an entry in alloc_order_map_
+  //   3. has the highest number of non_broadcast IterDomain
+  //
+  // The function is used to resolve allocation order propagation for operator
+  // with multiple operands. The one operand with the most number of
+  // non-broadcast IterDomain will be dominating the output allocation order.
+  // The motivation behind it to avoid breaking allocation order propagation
+  // from operands produced by broadcast. e.g. When a binary operator could take
+  // in a channels_last 4d tensor and an unsqueezed bias vector. We'll want to
+  // propagate the channels_last allocation order to output.
+  TensorView* resolveAllocationOrder(const std::vector<Val*>& candidates);
```

Comment:
This is merely a refactor to remove the special short-cut for binary operation. Idea is basically the same that we want to propagate the *biggest* tensor's allocation order to output.